### ISSUE TITLE
[AllyAudit] Correct focus order for Masthead and Masthead Extended [III]

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/spark-design-system#3849]
**User Story**
As Spark we want to Correct focus order for Masthead and Masthead Extended so that they are accessible.

**Notes:**
- Paciello Audit on Spark Design System
- QLC-17 Masthead & Masthead Extended
- Incorrect focus order
- This is happening in HTML but not Angular or React

**ISSUE DESCRIPTION**
Masthead and 79 - Masthead (extended) when the component is in a mobile layout, once the expandable dropdown components have been expanded once, their corresponding list of links remain in the focus order, even with the lists collapsed. This leads to a visual loss of focus. The expected tab order would be from one "Navigation Item" to the next but instead there are additional tab stops that are not visually apparent for the list of links not in view.
Nominally, these components also fail to have an indication of visible focus for the visually hidden elements that are still in the focus order. However, these elements should not be in the focus order, and once this is addressed the visible focus indication issue will no longer be applicable.

**RECOMMENDATIONS**
We recommend applying the approach outlined in Inaccessible custom dropdown (QLC-15) to refactor these dropdown widgets. If the lists are hidden using the CSS display:none property, the links will not be in the focus order.

**AC**
- Narrow Masthead Layout: when items are hidden in the masthead navigation tab a user will not be able to tab to them.

**Comment by ssabo1** [comment]
Confirmed: This has been added to Spark Design System Team Backlog.
